### ISSUE TITLE
Training Flashzoi

### ISSUE THREAD

**Issue by Jaureguy760** [johahi/borzoi-pytorch#11]
Hi again :) 

Just a quick question regarding the flashzoi training. In the preprint, its stated the batch norm is frozen, but it also states that the model trunk for human was fixed with the original weights and the mouse head and the transformer blocks were trained. I wanted to give the pretraining a shot. 

Questions:
1)Is this equivalent to what was done in scooby? or is this more direct freezing the head layers in the model? I know batch norm was frozen, but I was seeking further clarification about the human trunk component. https://github.com/gagneurlab/scooby/blob/37bdb2fd51097b3cba8fb63f2cdbe61e0b9f38c1/scooby/modeling/scooby.py#L37

2)How long did the training run take on 8 a40s with flash attention 2? 
3)Did you have any scripts showing the model freezing? 
4)Data augmentation was not mentioned in the preprint. Did you apply data augmentation similarly to the original borzoi model such as augment shift and rc like was done in scooby?

**Comment by johahi** [comment]
Hi again :) 
1. The param freezing is not equivalent. Scooby freezes Borzoi completely, and then adds and trains LoRA-matrices (+ the transform_borzoi_emb and the cell_state_conv). For Flashzoi training, we freeze everything but the mouse head and the transformer (Unet-bottleneck). The latter gets re-initialized with blocks consisting of rotary pos + FA2.
2. Training Flashzoi when partially initialized from Borzoi takes around 3hrs on 8 A40s.
3. You mean this?
```python
    for name,param in flashzoi.named_parameters():
        if "transformer" in name:
            accelerator.print (name, param.requires_grad, param.shape)
            continue
        elif "head" in name and "mouse" in name:
            accelerator.print (name, param.requires_grad)
            continue
        else: 
            param.requires_grad = False
```
And this for batchnorm:
```python
batchnorm_modules = [module for module in flashzoi.modules() if isinstance(module, torch.nn.BatchNorm1d)]
[module.eval() for module in batchnorm_modules]
```
4. Thanks for catching that! Indeed, sequences are randomly reverse complemented and shifted up to 3bp left/right. Will add this to the next version of the manuscript.
